src/models/Order.js

```diff
@@ -97,23 +104,23 @@ orderSchema.pre('find', async function (next) {
 });
 
 // Middleware: Update order statuses before executing a findOne operation
-orderSchema.pre('findOne', async function (next) {
+OrderSchema.pre('findOne', async function (next) {
   console.log('Pre-findOne started');
   // Call the updateOrderStatus function to update order statuses
   await updateOrderStatus();
   console.log('Pre-findOne finished');
   next();
 });
-// TTL (Time To Live) index to automatically delete orders with "cancelled" status after 2 minutes
-orderSchema.index(
+// TTL (Time To Live) index to automatically delete orders with "cancelled" status after 2 hours
+OrderSchema.index(
   { expiresAt: 1 },
   {
-    expireAfterSeconds: 120,
+    expireAfterSeconds: 7200,
```

Comment:
:white\_check\_mark:

Reply:
@akosasante does this mean that two hours after the order being cancelled this order will be deleted from the db? so if we are in dev mode and the status of unpaid order will change in 20h so 22h later it will be deleted from mongo, right? What does { expiresAt: 1 } mean in this case? in mongo we have 
createdAt
2023-08-08T20:45:00.942+00:00
updatedAt
2023-08-17T10:10:42.861+00:00 but can't find expiresAt- where did it come from?

Reply:
are we sorting all orders in mongo db depending on it's time that it has a cancelled status? basically putting the oldest one first to delete it first ?

Reply:
> does this mean that two hours after the order being cancelled this order will be deleted from the db? so if we are in dev mode and the status of unpaid order will change in 20h so 22h later it will be deleted from mongo, right?

Yes, that's correct.

> What does { expiresAt: 1 } mean in this case? in mongo we have

I'm not super familiar with Mongo + expiry/TTL (time-to-live) but my understanding from reviewing Marina's ticket is that it is basically a boolean. So 0 = false, an 1 = true. When true, it tells Mongo this document can be deleted. But since it's part of this index, this statement is saying set `expiry` flag on this document to `true` after the `expireAfterSeconds` time has passed. And the `partialFilterExpression` means this should only happen to orders with an `orderStatus` of `'cancelled'`

Reply:
Looking at : [https://www.mongodb.com/docs/manual/core/index-ttl/](https://www.mongodb.com/docs/manual/core/index-ttl/) and [https://www.mongodb.com/docs/manual/core/index-partial/](https://www.mongodb.com/docs/manual/core/index-partial/) it may be more accurate to say that yes we're creating a sorting of documents by this hidden field `expiresAt` but otherwise what I said above should still apply

Reply:
@akosasante got it!!! thanks for the explanations!!!